src/main/java/com/google/devtools/build/lib/skylarkbuildapi/repository/SkylarkRepositoryContextApi.java

```diff
@@ -294,6 +294,31 @@ public SkylarkExecutionResultApi execute(
   public boolean delete(Object path, Location location)
       throws EvalException, RepositoryFunctionExceptionT, InterruptedException;
 
+  @SkylarkCallable(
+      name = "patch",
+      doc =
+          "Apply a patch file to the external repository.",
+      useLocation = true,
+      parameters = {
+          @Param(
+              name = "patch_file",
```

Comment:
Should we also support inline patching? Having "patch_file_contents" attribute?

Reply:
> But I think we can just implement this in the Starlark repository rule instead of having a more complicated API?
In that case, we will need to create a temporary file and write the patch contents there.
I suspect that temporary file will have to be under the repository root, because of the repository context api limitations.
But you are right, this is additional functionality.
I wrote it because we already have similar thing for build_file_content attribute.